lib/mindwendel/lanes.ex

```diff
@@ -78,17 +79,33 @@ defmodule Mindwendel.Lanes do
           asc: lane.inserted_at
         ]
 
-    Repo.all(lane_query)
+    ideas_query = Idea
+    ideas_advanced_query = build_ideas_query_with_filter(ideas_query, filters[:idea_labels])
+
+    lane_query
+    |> Repo.all()
     |> Repo.preload(
-      ideas: [
-        :link,
-        :likes,
-        :label,
-        :idea_labels
-      ]
+      ideas:
+        ideas_advanced_query
+        |> preload([
+          :link,
+          :likes,
+          :idea_labels
+        ])
     )
   end
 
+  defp build_ideas_query_with_filter(query, nil) do
+    query
+  end
+
+  defp build_ideas_query_with_filter(query, _) do
+    from idea in query,
```

Comment:
Still don't think you need `ideas_query`/`query` argument. The one above can just return `Idea` and this one can just start with `from idea in Idea`.

You might want to make query manipulation for filters recursive from here on or maybe not, but that can be handled just here